nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RecordStats.java

```diff
@@ -0,0 +1,165 @@
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+        .name("record-stats-reader")
+        .displayName("Record Reader")
+        .description("A record reader to use for reading the records.")
+        .addValidator(Validator.VALID)
+        .identifiesControllerService(RecordReaderFactory.class)
+        .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("If a flowfile is successfully processed, it goes here.")
+        .build();
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("If a flowfile fails to be processed, it goes here.")
+        .build();
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+            .name(propertyDescriptorName)
+            .displayName(propertyDescriptorName)
+            .dynamic(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    }
+
+    private RecordPathCache cache;
+
+    @OnScheduled
+    public void onEnabled(ProcessContext context) {
+        cache = new RecordPathCache(25);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return new HashSet<Relationship>() {{
+            add(REL_SUCCESS);
+            add(REL_FAILURE);
+        }};
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if (input == null) {
+            return;
+        }
+
+        try {
+            Map<String, RecordPath> paths = getRecordPaths(context);
+            Map<String, String> stats = getStats(input, paths, context, session);
+
+            input = session.putAllAttributes(input, stats);
+
+            session.transfer(input, REL_SUCCESS);
+
+        } catch (Exception ex) {
+            getLogger().error("Error processing stats.", ex);
+            session.transfer(input, REL_FAILURE);
+        }
+
+    }
+
+    protected Map<String, RecordPath> getRecordPaths(ProcessContext context) {
+        return context.getProperties().keySet()
+            .stream().filter(p -> p.isDynamic() && !p.getName().contains(RECORD_READER.getName()))
+            .collect(Collectors.toMap(
+                e -> e.getName(),
+                e ->  {
+                    String val = context.getProperty(e).getValue();
+                    return cache.getCompiled(val);
+                })
+            );
+    }
+
+    protected Map<String, String> getStats(FlowFile input, Map<String, RecordPath> paths, ProcessContext context, ProcessSession session) {
+        try (InputStream is = session.read(input)) {
+            RecordReaderFactory factory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
```

Comment:
input and is as var names is confusing here, can we name them closer to what they are to keep them straight?
flowFile and inputStream?
There is only one flowFile to track in this processor so none to get fancy ;)

Reply:
Done.